### PR TITLE
Push db_table param into SimpleTable

### PR DESCRIPTION
**Pull request by longhotsummer**
@xybrnet

### CODE REVIEW

File: wazimap/data/tables.py [hunk 1]
```diff
@@ -626,16 +628,16 @@ def _table_name(self, geo_level=None):
...
                 raise ValueError('Expected a geo_level')
-            return '%s_%s' % (self.id, geo_level)
+            return '%s_%s' % (self.db_table, geo_level)
 
-        return self.id
+        return self.db_table
 
 
-_census_table_models = {}
+_field_table_models = {}
```
Comment (xybrnet) [inline wazimap/data/tables.py]: I also had an urge to change this.

File: wazimap/data/tables.py [hunk 2]
```diff
@@ -603,9 +605,9 @@ def _build_model_from_fields(self, fields, table_name, geo_level=None, value_typ
 
         # create the table model
         class Model(Base):
-            table = existing_table or table_name
+            table = table_name
             __table__ = Table(table, Base.metadata, *table_args, extend_existing=True)
-        _census_table_models[table_name] = Model
+        _field_table_models[table_name] = Model
```
Comment (xybrnet) [inline wazimap/data/tables.py]: I don't think this is going to work.
e.g. if we have `pop_group_by_gender` and `gender_by_pop_group` using the same table, `gender_by_pop_group` , we need both keys in `_field_table_models`. In the above only `gender_by_pop_group` will be in there.

File: wazimap/data/tables.py (hunk 1 above)
Comment (longhotsummer) [reply wazimap/data/tables.py]: :)

**Comment by longhotsummer** [comment]
I've made additional changes so that we're smarter about whether we're using the database table name, or the FieldTable name, when looking up tables. In almost all places, especially when supporting the profiles, we use the FieldTable name.

File: wazimap/data/tables.py [hunk 3]
```diff
@@ -355,8 +355,8 @@ def __init__(self, fields, id=None, universe='Population', description=None, den
         self.value_type = getattr(sqlalchemy.types, value_type)
 
         if db_table:
-            model = get_model_by_name(db_table)
-            if set(fields) != set(model.data_table.fields):
+            model = get_model_for_db_table(db_table)
+            if model and set(fields) != set(model.data_table.fields):
```
Comment (xybrnet) [inline wazimap/data/tables.py:359]: Nice.

File: wazimap/data/tables.py [hunk 4]
```diff
@@ -605,9 +607,8 @@ def _build_model_from_fields(self, fields, table_name, geo_level=None, value_typ
 
         # create the table model
         class Model(Base):
-            table = table_name
+            table = db_table
             __table__ = Table(table, Base.metadata, *table_args, extend_existing=True)
```
Comment (xybrnet) [inline wazimap/data/tables.py]: We could probably just do:
`Table(db_table, Base.metadata, *table_args, extend_existing=True)`

File: wazimap/data/tables.py [hunk 5]
```diff
@@ -633,40 +634,46 @@ def _table_name(self, geo_level=None):
         return self.db_table
 
 
-_field_table_models = {}
+def get_model_for_db_table(db_table):
+    """ Lookup the SQLAlchemy model for a particular database table.
+    """
+    for model in Base._decl_class_registry.values():
+        if hasattr(model, '__tablename__') and model.__tablename__ == db_table:
+            return model
```
Comment (xybrnet) [inline wazimap/data/tables.py:642]: Awesome - this makes it much more clear!

File: wazimap/data/tables.py (hunk 4 above)
Comment (longhotsummer) [reply wazimap/data/tables.py]: Done!

**Comment by xybrnet** [comment]
👍

**Pull request merged**